### DISCUSSION TITLE
128bit Service UUID

### DISCUSSION THREAD

**Discussion by ClackeyLHD** [h2zero/NimBLE-Arduino#600] [Q&A]
Hi,
I tried to replace the "BAAD" UUID in the server example to create a service with the 128bit UUID given in the [documentation](https://h2zero.github.io/esp-nimble-cpp/class_nim_b_l_e_u_u_i_d.html#a399dca7d70ee262aa155b39d357321b1) as follows:

NimBLEService* pBaadService = pServer->createService(0xebe0ccb0, 0x7a0a, 0x4b0c, 0x8a1a6ff2997da3a6);

This is resulting in Compilation error: no matching function for call to 'NimBLEServer::createService(unsigned int, int, int, long long unsigned int)'

How can I correct this? Thanks!

**Comment by h2zero** [comment]
You'll need to specify the type for your hex values or else they default to int `(uint16/32/64_t) 0xffff` etc.

**Comment by ClackeyLHD** [comment]
Thanks!
